deploy/helm/sumologic/conf/traces/traces.otelcol.conf.yaml

```diff
@@ -4,71 +4,47 @@ receivers:
     protocols:
       grpc:
   zipkin:
-    endpoint: "0.0.0.0:9411"
+    endpoint: {{ .Values.otelcol.receivers.zipkin.endpoint }}
 processors:
   k8s_tagger:
-    passthrough: false
+    passthrough: {{ .Values.otelcol.processors.k8s_tagger.passthrough }}
     extract:
       metadata:
-        # extract the following well-known metadata fields
-        - containerId
-        - containerName
-        - containerImage
-        - cluster
-        - daemonSetName
-        - deployment
-        - hostName
-        - namespace
-        - namespaceId
-        - node
-        - owners
-        - podId
-        - podName
-        - replicaSetName
-        - serviceName
-        - startTime
-        - statefulSetName
+        {{- range .Values.otelcol.processors.k8s_tagger.extract.metadata }}
+        - {{ . | quote }}
+        {{- end }}
       tags:
-        containerId: container_id
-        containerName: container_name
-        containerImage: conatiner_image
-        cluster: cluster
-        daemonSetName: daemonset
-        deployment: deployment
-        hostName: hostname
-        namespace: namespace
-        namespaceId: namespace_id
-        node: node
-        podId: pod_id
-        podName: pod
-        replicaSetName: replicaset
-        serviceName: service_name
-        startTime: start_time
-        statefulSetName: statefulset
+        {{- range $key, $val := .Values.otelcol.processors.k8s_tagger.extract.tags }}
+        {{ $key }}: {{ $val | quote }}
+        {{- end }}
       annotations:
-         - tag_name: pod_annotation_%s
-           key: "*"
+        {{- range .Values.otelcol.processors.k8s_tagger.extract.annotations }}
+        - tag_name: {{ .tag_name | quote }}
+          key: {{ .key | quote }}
+        {{- end }}
       labels:
-         - tag_name: pod_label_%s
-           key: "*"
+        {{- range .Values.otelcol.processors.k8s_tagger.extract.labels }}
+        - tag_name: {{ .tag_name | quote }}
+          key: {{ .key | quote }}
+        {{- end }}
   queued_retry:
-    num_workers: 16
-    queue_size: 10000
-    retry_on_failure: true
+    num_workers: {{ .Values.otelcol.processors.queued_retry.num_workers }}
+    queue_size: {{ .Values.otelcol.processors.queued_retry.queue_size }}
+    retry_on_failure: {{ .Values.otelcol.processors.queued_retry.retry_on_failure }}
   batch:
-    send_batch_size: 1024
-    timeout: 5s
+    send_batch_size: {{ .Values.otelcol.processors.batch.send_batch_size }}
+    timeout: {{ .Values.otelcol.processors.batch.timeout }}
```

Comment:
@perk-sumo The challenge here is that there is a number of processors: https://github.com/open-telemetry/opentelemetry-collector/tree/master/processor - the list is actually growing and their properties might change over time.

An ideal solution would to provide a template with the full config file, but we are not sure how to tackle it. Do you have some suggestions?

Reply:
@sumo-drosiek perhaps we should minimally add sampling processors: https://github.com/open-telemetry/opentelemetry-collector/tree/master/processor#sampling-processor ?

Reply:
As I think about it more and more, we should just take full config branch and put it as a configmap. If user have own processors it's gonna add changes to values or config file anyway. Additionally we want to support `--set` flags

@perk-sumo any thoughts?

Reply:
not so nice proposal, but had to support dynamic value for url :D

https://github.com/SumoLogic/sumologic-kubernetes-collection/commit/bb9c9b08b6a8cffd2edbf2797918da737d131b4f

Reply:
I'd like to keep some defaults for clients who don't want to deal with the config that much.
We could do it like fluent-bit: provide sane defaults in config.yaml file with some flags in values.yaml and if someone want to have a totally different config provide a special key in values.yaml for that.
We could also provide the template config but I'm not sure if that should go to values.yaml or we should create a new file for that like values.otelcol.yaml or sth. With that client would need to copy and paste something into values.yaml by hand.
I don't like the idea of whole otelcol config inside default values.yaml because it will grow really huge and that doesn't seem right.

Please take a look at:
https://github.com/helm/charts/blob/master/stable/fluent-bit/values.yaml#L146
https://github.com/helm/charts/blob/master/stable/fluent-bit/templates/config.yaml#L1

It's similar to what @sumo-drosiek did.

Reply:
`We could do it like fluent-bit` I think that our case differs. We are able to use `yaml`. fluentbit were obligated to convert yaml configuration into fluent-bit specific format.
I would like to stick to the `config.yaml`. It's more clear and in both cases `values.yaml` will be growing